_posts/2019-03-13-activeadmin-vs-administrate.md

```diff
@@ -0,0 +1,41 @@
+
+# Enter our contenders
+
+Since they're both implemented as Rails engines, to a large extent using either Administrate or ActiveAdmin will make the choice for you - you'll start out leaning heavily towards the separate application camp.  I make a note of that because, if you really want to just create an administrative overlay on your existing application, then neither one of these options will help you much.
+
+# ActiveAdmin
+
+If you're a big fan of DSLs, then take a look at  [ActiveAdmin](https://activeadmin.info).  The core pattern behind it is a straightforward DSL that lets you register a resource, and then add it's fields to a largely pre-defined interface.
+
+Between the two options, ActiveAdmin generally takes less work to stand up a running application, and provides much better filtering and sorting out of the box.  This is also a personal opinion, but the UI seems more polished.  However, I've found customizing the interface with additional javascript and UI flair like inline editing and dropdown menus to be more difficult.  That said, it's definitely a solid option, and one that you should lean towards if you just want to put up an admin app and things like filtering and sorting are really important for your use case.
+
+# Administrate
+
+If you prefer to stick to plain old Ruby and use Rails conventions, then [Administrate](https://administrate-prototype.herokuapp.com/getting_started) may be what you're looking for.  In keeping with general Rails philosophy, it adheres to a set of conventions to provide a pretty standard, "pre-canned" administrative application with minimal configuration needed.  However, if you need to extend or alter the standard functionality, then you can override the gem provided controllers, views, and styling elements using generators provided by the gem.  Just generate whatever you need to customize (views, controller actions), either for the app as a whole or for a specific resource, and edit away.  This makes Administrate more straightforward to customize if you have some more specific requirements.  This is good in general, and also (another personal opinion here) helps to make up for a rougher and less fully featured default interface.
+
+One drawback to be aware of is that, as of the time of writing this article,  Administrate does not support namespaced models.  Generating an install against an application with them will output a warning about this, and a console that makes no reference to any of the namespaced models.  Depending on how you've organized your application, that could potentially be a show stopper.
```

Comment:
@rdormer Not sure if it worth to dedicate a paragraph on that. It's not that common to fall into this issue.

Reply:
I tested out both of these gems by adding an administrative interface to a personal project of mine, and this was literally the first issue I ran into while using Administrate.

Reply:
I agree with @lubc. I believe a bigger drawback is that `administrate` is much newer than `activeadmin` so it is not as stable as AA. Their version is even pre-1.0.0